xrpl/models/transactions/transaction.py

```diff
@@ -371,6 +371,24 @@ def has_flag(self: Transaction, flag: int) -> bool:
         else:  # is List[int]
             return flag in self.flags
 
+    def is_signed(self: Transaction) -> bool:
+        """
+        Checks if a transaction has been signed.
+
+        Returns:
+            Whether the transaction has been signed
+        """
+        if self.signers:
+            for signer in self.signers:
+                if (
+                    signer.signing_pub_key is None or len(self.signing_pub_key) <= 0
+                ) and (signer.txn_signature is None or len(signer.txn_signature) <= 0):
+                    return False
+            return True
+        return (self.signing_pub_key is not None and len(self.signing_pub_key) > 0) or (
+            self.txn_signature is not None and len(self.txn_signature) > 0
```

Comment:
I think that both of these fields have to be there for it to be properly signed, but please double check - https://xrpl.org/transaction-common-fields.html#signers-field

```suggestion
        return (self.signing_pub_key is not None and len(self.signing_pub_key) > 0) and (
            self.txn_signature is not None and len(self.txn_signature) > 0
```

Reply:
I'd recommend adding some test cases to make sure the logic works out as expected.

Reply:
@JST5000 both of these fields should be there. xrpl.js PR fix:
https://github.com/XRPLF/xrpl.js/pull/2308
I will update it once we are good with the js PR and add some tests.

Reply:
Logic modified and tests added